### DISCUSSION TITLE
How to get_battery data without delay

### DISCUSSION THREAD

**Discussion by sk1kgn** [doronz88/pymobiledevice3#1201] [Q&A] [answered]
When I use DiagnosticsService.get_battery to monitor phone's battery current, the data was not update per second.
code like this: 
data = DiagnosticsRelayService(udid=udid, logger=logger).get_battery()
![image](https://github.com/user-attachments/assets/b8db4ef4-c723-4825-8b56-1566b11f085d)

By analyzing the data['UpdateTime'], it seems the data update per 20 second.  It's is so insensitive. 

So, is there any way to increase update frequency, like update per second?

Looking forward to your reply, thanks so mach.

**Comment by doronz88** [comment, marked as answer]
We don't have any control over the IORegistry values. We can only read, not write. From my understanding, this is exactly the time when the hardware notifies the OS about a change in one of the properties.
